Code/src/main/java/nl/utwente/group10/ghcj/GhciSession.java

```diff
@@ -55,6 +59,33 @@ public String pull(final Expr expr) throws HaskellException {
     }
 
     /**
+     * Returns the result of evaluating something in ghci.
+     * Should only be used for testing purposes or for a known valid Haskell expression. 
+     * @param expr The string representation of the expression to evaluate.
+     * @return The result of the evaluation.
+     * @throws HaskellException when ghci encountered an error.
+     */
+    public String pullRaw(final String expr) throws HaskellException {
+        return this.ghci.eval(expr).trim();
+    }
+    
+    /**
+     * Ask ghci for the type of an expression
+     * @param expr The expression String to determine the type of.
+     * @param env The environment in which the type will be resolved 
+     * @return The parsed Haskell type
+     * @throws HaskellException when  ghci encountered an error or the type could not be parsed.
+     */
+    public Type pullType(final String expr, Environment env) throws HaskellException {
+        String[] parts = this.pullRaw(":t (" + expr + ")").split(" :: ");
```

Comment:
There's an extra pair of parentheses around the expression here - do we also need that in push, above?